sympy/solvers/solvers.py

```diff
@@ -2653,22 +2653,54 @@ def _tsolve(eq, sym, **flags):
-                # f(x)**g(x) only has solutions where f(x) == 0 and g(x) != 0 at
-                # the same place
-                sol_base = _solve(lhs.base, sym, **flags)
-                if not sol_base:
-                    return sol_base  # no solutions to remove so return now
-                return list(ordered(set(sol_base) - set(
-                    _solve(lhs.exp, sym, **flags))))
-            elif (rhs is not S.Zero and
-                        lhs.base.is_positive and
-                        lhs.exp.is_real):
-                return _solve(lhs.exp*log(lhs.base) - log(rhs), sym, **flags)
-            elif lhs.base == 0 and rhs == 1:
-                return _solve(lhs.exp, sym, **flags)
             else:
-                raise NotImplementedError
+                # sym in lhs.exp.free_symbols
+                if not rhs:
+                    # f(x)**g(x) only has solutions where f(x) == 0 and g(x) != 0 at
+                    # the same place
+                    sol_base = _solve(lhs.base, sym, **flags)
+                    if not sol_base:
+                        return sol_base  # no solutions to remove so return now
+                    return list(ordered(set(sol_base) - set(
+                    _solve(lhs.exp, sym, **flags))))
+                elif lhs.base.is_positive and lhs.exp.is_real:
+                    return _solve(lhs.exp*log(lhs.base) - log(rhs), sym, **flags)
+                elif rhs.is_Integer:
+                    if rhs==1:
+                        if  lhs.base == 0:
+                            return _solve(lhs.exp, sym, **flags)
+                        else:
+                            # 1 = foo**0, 1**foo, (-1)**even
+                            b1 = _solve(lhs.base - 1, sym, **flags)
+                            b_1 = _solve(lhs.base + 1, sym, **flags)
+                            e0 = _solve(lhs.exp, sym, **flags)
+                            e1=[]
+                            rewrite = lhs.rewrite(exp)
+                            if rewrite != lhs:
+                                e1= _solve(rewrite - rhs, sym, **flags)
+                            sol = list(b1) + list(e0) + list(e1)
+                            for b in b_1:
+                                    if eq.subs(sym, b)==0:
+                                        sol.append(b)
+                            return list(set(ordered(sol)))
+                    else:
+                        sol=[]
+                        from sympy import divisors
+                        x=divisors(rhs)
+                        for i in x:
+                            if eq.subs(sym,i)==0:
+                                sol.append(i)
+                            elif eq.subs(sym,-i)==0:
+                                sol.append(-i)
+                        e1=[]
+                        rewrite = lhs.rewrite(exp)
+                        if rewrite != lhs:
+                            e1= _solve(rewrite - rhs, sym, **flags)
```

Comment:
@oscarbenjamin , @smichr but here i am getting not implemented error.the other positive solution i.e ```1.299``` is unabled to solve . I think it is not perfect lambert form to give solution as in case of ```x**x+1=0```. Moreover in this equation the rhs is turning out to be ```_t```  dummy variable. What do you think about above implementation?

Reply:
Wolfram only gives that root as a numerical solution. Maybe it doesn't have a form that we can write down symbolically. I think that's fine though - better to return some roots than none.